src/ConfigurationRemotingServer/Program.cs

```diff
@@ -3,36 +3,95 @@
 using Microsoft.Management.Configuration.Processor;
 using System.Reflection;
 using System.Runtime.InteropServices;
+using System.Runtime.Loader;
 using WinRT;
 
 namespace ConfigurationRemotingServer
 {
+    /// <summary>
+    /// Custom assembly load context.
+    /// </summary>
+    internal class NativeAssemblyLoadContext : AssemblyLoadContext
+    {
+        private static readonly string PackageRootPath;
+
+        private static readonly NativeAssemblyLoadContext NativeALC = new();
+
+        static NativeAssemblyLoadContext()
+        {
+            var self = typeof(NativeAssemblyLoadContext).Assembly;
+            PackageRootPath = Path.Combine(
+                Path.GetDirectoryName(self.Location)!,
+                "..");
+        }
+
+        private NativeAssemblyLoadContext()
+            : base("NativeAssemblyLoadContext", isCollectible: false)
+        {
+        }
+
+        /// <summary>
+        /// Handler to resolve unmanaged assemblies.
+        /// </summary>
+        /// <param name="context">Assembly load context.</param>
+        /// <param name="name">Assembly name.</param>
+        /// <returns>The assembly, null if not in our assembly location.</returns>
+        internal static IntPtr ResolvingUnmanagedHandler(Assembly context, string name)
+        {
+            if (name.Equals("WindowsPackageManager.dll", StringComparison.OrdinalIgnoreCase))
+            {
+                return NativeALC.LoadUnmanagedDll(name);
+            }
+
+            return IntPtr.Zero;
+        }
+
+        /// <inheritdoc/>
+        protected override IntPtr LoadUnmanagedDll(string unmanagedDllName)
+        {
+            string path = Path.Combine(PackageRootPath, unmanagedDllName);
+            if (File.Exists(path))
+            {
+                return this.LoadUnmanagedDllFromPath(path);
+            }
+
+            return IntPtr.Zero;
+        }
+    }
+
     internal class Program
     {
         static int Main(string[] args)
         {
-            ulong memoryHandle = ulong.Parse(args[0]);
+            // Help find WindowsPackageManager.dll
+            AssemblyLoadContext.Default.ResolvingUnmanagedDll += NativeAssemblyLoadContext.ResolvingUnmanagedHandler;

```

Comment:
This was required only for the elevation scenario; for some reason it would not find the unmanaged dll any more.  I don't understand why, but c'est la vie.